starfyre/parser.py

```diff
@@ -162,35 +170,39 @@ def handle_endtag(self, tag):
         endtag_node = self.stack.pop()  
         self.current_depth -= 1
 
+
         if endtag_node.tag != "style" and endtag_node.tag != "script":
             if len(self.stack) > 0:
-                parent_node = self.stack[-1]      #this is last item/"top element" of stack
-
-                if endtag_node.original_name != endtag_node.tag: #this node is for "custom" component
-                    #process endtag_node children                    
-                    new_children = []
-                    is_slot_used = False
-                    for child_component in endtag_node.children:                        
-                        if child_component.is_slot_component:
-                            new_children.extend(self.current_children)
-                            is_slot_used = True
-                        else:
-                            new_children.append(child_component)
-                    if not is_slot_used and len(self.current_children) > 0:
+                parent_node = self.stack[-1]  # this is last item/"top element" of stack
+                print("This is the end tag node", endtag_node.original_name, self.current_children)
+
+                # Process endtag_node children, since there's no check for original_name
+                new_children = []
+                is_slot_used = False
+                for child_component in endtag_node.children:                        
+                    if child_component.is_slot_component:
                         new_children.extend(self.current_children)
```

Comment:
@SuelenKarbivnychyy , we should investigate this line. This is the line that determines the order of rendering of slot elements.

Reply:
I'm on it :)